azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/util/DeploymentTemplateBuilder.java

```diff
@@ -0,0 +1,356 @@
+              .name("[variables('networkInterfaceCardName')]")
+              .type("Microsoft.Network/networkInterfaces")
+              .location(location)
+              .apiVersion("2015-06-15")
+              .dependsOn(Arrays.asList("[concat('Microsoft.Network/publicIPAddresses/', variables('publicIPAddressName'))]",
+                      "[concat('Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'))]"))
+              .properties(networkInterfaceCardProperties)
+              .build();
+
+      resources.add(networkInterfaceCard);
+   }
+
+   private void addVirtualMachine() {
+
+      //Build OS Profile
+      final String computerName = name + "pc";
+
+      variables.put("loginUser", loginUser);
+      OSProfile.Builder profileBuilder = OSProfile.builder()
+              .adminUsername(loginUser)
+              .computerName(computerName);
+
+      boolean usePublicKey = options.getPublicKey() != null;
+
+      if (usePublicKey) {
+         OSProfile.LinuxConfiguration configuration = OSProfile.LinuxConfiguration.create("true",
+                 OSProfile.LinuxConfiguration.SSH.create(Arrays.asList(
+                         OSProfile.LinuxConfiguration.SSH.SSHPublicKey.create(
+                                 "[concat('/home/',variables('loginUser'),'/.ssh/authorized_keys')]",
+                                 options.getPublicKey())
+                 ))
+         );
+         profileBuilder.linuxConfiguration(configuration);
+      } else {
+         profileBuilder.adminPassword(loginPassword);
+      }
+
+      OSProfile osProfile = profileBuilder.build();
+
+      //Build Image Reference
+      final String imagePublisher = template.getImage().getProviderId();
+      final String imageOffer = template.getImage().getName();
+      final String imageSku = template.getImage().getVersion();
+
+      ImageReference imageReference = ImageReference.builder()
+              .publisher(imagePublisher)
+              .offer(imageOffer)
+              .sku(imageSku)
+              .version("latest")
+              .build();
+
+      //Build OsDisk
+      final String storageAccountContainerName = "vhds";
+      variables.put("storageAccountContainerName", storageAccountContainerName);
+
+      final String osDiskName = name + "osdisk";
+      variables.put("osDiskName", osDiskName);
+
+      OSDisk osDisk = OSDisk.builder()
+              .name("[variables('osDiskName')]")
+              .vhd(
+                      VHD.create("[concat('http://',variables('storageAccountName'),'.blob.core.windows.net/',variables('storageAccountContainerName'),'/',variables('osDiskName'),'.vhd')]")
+              )
+              .caching("ReadWrite")
+              .createOption("FromImage")
+              .build();
+
+
+      //Create Data Disk(s) and add to list
+      final String dataDiskName = name + "datadisk";
+      variables.put("dataDiskName", dataDiskName);
+      final String dataDiskSize = "100";
```

Comment:
Hi, Yes eventually data disk information (number of disks and their sizes) should be set using template option. In Azure the number of supported data disks depends on hardware in use, but the max size for singe disk is 1023 GB. Should we just drop data disk creation until Azure template option has support for it or create a singe data disk using 1023 as default size?